tests/e2e/vmc_csi_deployments.go

```diff
@@ -0,0 +1,215 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	dep "k8s.io/kubernetes/test/e2e/framework/deployment"
+	fpv "k8s.io/kubernetes/test/e2e/framework/pv"
+)
+
+var _ = ginkgo.Describe("[vmc-gc] [csi-guest] Deploy Update and Scale Deployments", func() {
+	f := framework.NewDefaultFramework("e2e-vsphere-deployment")
+	var (
+		namespace         string
+		client            clientset.Interface
+		storagePolicyName string
+		scParameters      map[string]string
+		deploymentName    string
+		replica           int32
+		maxReplica        int32
+		pvclaims          []*v1.PersistentVolumeClaim
+		pvclaim           *v1.PersistentVolumeClaim
+		sc                *storagev1.StorageClass
+		svNamespace       string
+	)
+	ginkgo.BeforeEach(func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		namespace = getNamespaceToRunTests(f)
+		client = f.ClientSet
+		bootstrap()
+
+		ginkgo.By("Set Resource Quota")
+		svcClient, svNamespace = getSvcClientAndNamespace()
+		setResourceQuota(svcClient, svNamespace, rqLimit)
+
+		sc, err := client.StorageV1().StorageClasses().Get(ctx, storageclassname, metav1.GetOptions{})
+		if err == nil && sc != nil {
+			gomega.Expect(client.StorageV1().StorageClasses().Delete(ctx, sc.Name, *metav1.NewDeleteOptions(0))).NotTo(gomega.HaveOccurred())
+		}
+		scParameters = make(map[string]string)
+		storagePolicyName = GetAndExpectStringEnvVar(envStoragePolicyNameForSharedDatastores)
+	})
+
+	ginkgo.AfterEach(func() {
+		ginkgo.By("Modify Resource Quota")
+		svcClient, svNamespace := getSvcClientAndNamespace()
+		setResourceQuota(svcClient, svNamespace, defaultrqLimit)
+
+		if pvclaim != nil {
+			err := fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+	})
+
+	/*
+		Test performs following operations
+
+		Steps
+		1. Deploy nginx pods with replica count as 3
+		2. Update the nginx deployment pods, update the nginx image from k8s.gcr.io/nginx-slim:0.8 to k8s.gcr.io/nginx-slim:0.9
+		3. Wait for some time and verify the update is successful
+		4. Scale up the deployment to 10 replicas and verify all the pods should be up and running
+		5. Scale down the deployment to 3 replicas and verify all the pods should be up and running
+		6. Delete the deployments and Verify deployments are deleted successfully
+
+	*/
+
+	ginkgo.It("Scale up and scale down deployments", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		var err error
+		var lables = make(map[string]string)
+		lables["app"] = "nginx"
+		deploymentName = "testdeployment"
+		replica = 3
```

Comment:
If this is for block volume, then it will not work since block volume are RWO, same volume cannot be used by different pods(replicas).

Reply:
As discussed I have removed this testcase